presto-native-execution/presto_cpp/main/common/ConfigReader.cpp

```diff
@@ -36,10 +35,12 @@ std::unordered_map<std::string, std::string> readConfig(
       continue;
     }
 
-    std::vector<std::string> configParts;
-    folly::split('=', line, configParts);
-    VELOX_USER_CHECK_EQ(configParts.size(), 2, "Malformed config: {}", line);
-    properties.emplace(configParts[0], configParts[1]);
+    auto delimiterPos = line.find('=');
+    VELOX_CHECK_NE(delimiterPos, std::string::npos);
+    auto name = line.substr(0, delimiterPos);
+    VELOX_CHECK(!name.empty());
```

Comment:
Should we trim the 'name'?

Reply:
it's trimmed at the beginning of the loop

Reply:
Hmm... Let's say we have "   xxx = yyy  ". At the start of the loop we trim this to "xxx = yyy", then we get name="xxx " with space at the end, no?

Reply:
at the start of the loop we get xxx=yyy so we get name = "xxx"

Reply:
The original logic essentially takes out all spaces in the line before proceeding to parse the key and values. I assume the assumption was made that no spaces are allowed in keys and values.

Reply:
Got it. Thank you for clarifying. It would be nice to document this behavior.

Reply:
Thinking about it again, this logic seems dangerous. Looks like it allows

```
foo=bar
f o o = b  ar
```

Does it make sense to allow spaces in config names? CC: @amitkdutta @xiaoxmeng